model/sharing/indexer.go

```diff
@@ -293,7 +293,21 @@ func (s *sharingIndexer) UpdateDirDoc(olddoc, doc *vfs.DirDoc) error {
 	}
 	doc.SetRev(s.bulkRevs.Rev)
 	docs[0]["_rev"] = s.bulkRevs.Rev
-	docs[0]["_revisions"] = s.bulkRevs.Revisions
+	if olddoc != nil {
+		/*
+			XXX We cannot directly apply the revision chain as received by the remote
+			instance because it might create a CouchDB conflict if a rev does not
+			exist on the local instance. Therefore, we start from the last local rev
+			and apply all the higher revs received from the remote.
+		*/
+		chain := revsStructToChain(s.bulkRevs.Revisions)
+		newChain := MixupChainToResolveConflict(olddoc.DocRev, chain)
+		newRevs := revsChainToStruct(newChain)
+		docs[0]["_revisions"] = newRevs
+	} else {
+		docs[0]["_revisions"] = s.bulkRevs.Revisions
+	}
```

Comment:
[nit] I would prefer to write it as:

```go
newRevs := s.bulkRevs.Revisions
if olddoc != nil {
    // ...
    newRevs = revsChainToStruct(newChain)
}
docs[0]["_revisions"] = newRevs
```

I think it is a bit easier to modify later if needed. In particular, it avoids to repeat `docs[0]["_revisions"] =`.